claat/parser/md/parse_test.go

```diff
@@ -0,0 +1,339 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package md
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/googlecodelabs/tools/claat/types"
+
+	"golang.org/x/net/html"
+)
+
+const empty = ""
+
+var errNonNil = errors.New(empty)
+
+func buildParserWithStep(markup string) *parserState {
+
+	ps := parserState{
+		tzr: html.NewTokenizer(strings.NewReader(markup)),
+		c:   &types.Codelab{},
+	}
+	ps.currentStep = ps.c.NewStep("title")
+	return &ps
+}
+
+func TestHandleCodelabTitle(t *testing.T) {
+	// Set up.
+	title := "Egret"
+	ps := buildParserWithStep(fmt.Sprintf("<h1>%s</h1>", title))
+	ps.advance()
+	handleCodelabTitle(ps)
+
+	if ps.c.Title != title {
+		t.Errorf("[%q] expected %v, got %v", title, title, ps.c.Title)
+	}
+}
+
+func TestProcessDuration(t *testing.T) {
+	type TestCase struct {
+		in  string
+		out time.Duration
+		err error
+	}
+	tests := []TestCase{
```

Comment:
yeah, go fmt will expand struct{type1; type2; type3} into multiline, I just wrote it like that for brevity :) 
my main point was the format of the slice elements, e.g. `{"1:00", time.Hour, nil}`.